### ISSUE TITLE
Implement ConfigFlow

### ISSUE THREAD

**Issue by sbidy** [sbidy/wiz_light#4]
Add ConfiFlow to the integration to support UI based configuration parameter.

**Comment by LanceMcCarthy** [comment]
How can I help with this? Point me to the configflow docs and I will implement it.

**Comment by ChrisLizon** [comment]
@LanceMcCarthy Disclaimer, I'm not a developer, I'm just a hack 😁  I think this is the documentation you need. 

https://developers.home-assistant.io/docs/config_entries_config_flow_handler/ 

I am not sure if the Unique ID mechanism needs to put in place first, as that seems to be a prerequisite. The interesting part to me is where that says that properties should not do IO, but we need to get the device id from the device, so that's where my understanding falls apart. 

https://developers.home-assistant.io/docs/core/entity/#generic-properties

https://developers.home-assistant.io/docs/entity_registry_index#unique-id-requirements

I would assume @sbidy  was planning to use the last 6 digits of the mac address, as that is what the bulbs seem to use as their hostname. It's also returned by a `getSystemConfig` call.

**Comment by ChrisLizon** [comment]
also @LanceMcCarthy  I think @sbidy  has been busy on this already. 

https://github.com/sbidy/core/commits/dev/homeassistant/components/wiz_light

**Comment by sbidy** [comment]
Yes, I'm currently working on that to push this integration into the HA core as an standard.

I used the MAC as unique ID (as recommended in the HA docs). The ConfigFlow is also working in my fork but I have to do some testing and clean up.

I'm looking forward to create a new pull request to the HA core in the next few days.

**Comment by LanceMcCarthy** [comment]
Thanks! That looks promising.

I peeked at @sbidy fork's config_flow.py and see the wiz_light device name is an optional field (see  
https://github.com/sbidy/core/blob/e1ca3f9a297516e0902ca0dbb9ff297ea92902d1/homeassistant/components/wiz_light/config_flow.py#L20)

In that case, I would suggest prefixing the IP address with `wiz_light` to help prevent ID collisions? For example, if the bulb is at **192.168.1.100**, then the ID should be `wiz_light_192_168_1_100`. 

This would be more robust because other integrations can end up using raw IP addresses, too. For example, **iperf3** if you use an IP address instead of a domain name for a server, then the device ID will just be an IP address 

![image](https://user-images.githubusercontent.com/3520532/102351094-e4f54400-3f73-11eb-90fb-ae7cf52bdb8c.png)

**Comment by craigcabrey** [comment]
@sbidy are you still working on this? I started implementing it for this component (see initial work in https://github.com/craigcabrey/wiz_light/tree/config_flow). If you haven't gotten very far I'd be happy to take over this functionality :)

**Comment by sbidy** [comment]
@LanceMcCarthy I've added the devices with their MAC in the Domain of the wiz_light. This is defined as Best Practice for HA integrations. The ID must only be unique within the domain. The Entity Name can be defined by the user and must not be unique. See in the #dev Discord from HA.

@craigcabrey I've already added the config_flow and some other features in my dev fork from HA. The first Pull to the HA dev branch will initiated to day.
